app/components/blacklight/document/page_header_component.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module Document
+    # Render the start over and prev/next displays
+    class PageHeaderComponent < Blacklight::Component
+      attr_reader :document, :blacklight_config, :search_context, :search_session, :has_header_actions
+
+      def initialize(document:, blacklight_config:, search_context:, search_session:)
+        super
+        @search_context = search_context
+        @search_session = search_session
+        @blacklight_config = blacklight_config
+        @document = document
+      end
+
+      def render?
+        search_context.present? || search_session.present? || has_header_actions
+      end
+
+      def before_render
```

Comment:
why is this `before_render` Can we make a method `has_header_actions?` (and remove the `attr_reader`)

Reply:
done.